src/common/quantile.cc

```diff
@@ -435,6 +478,13 @@ void SketchContainerImpl<WQSketch>::MakeCuts(Context const *ctx, MetaInfo const
     p_cuts->cut_ptrs_.HostVector().push_back(cut_size);
   }
 
+  if (info.IsVerticalFederated() && info.IsSecure()) {
+      // cut values need to be synced across all workers via Allreduce
```

Comment:
wouldn't it leak feature values to other workers?

Reply:
to be handled by another PR